src/reactor/cache.js

```diff
@@ -0,0 +1,201 @@
+import { List, Map, OrderedSet, Record } from 'immutable'
+
+export const CacheEntry = Record({
+  value: null,
+  storeStates: Map(),
+  dispatchId: null,
+})
+
+/*******************************************************************************
+ * interface PersistentCache {
+ *    has(item)
```

Comment:
`lookup` and `has` are simply analogous to `get` and `has` in an `Immutable.Map`.

The reason why there seems to be extra operations is because nuclear is designed to be used with a persistent data structure for the cache. For a caching abstraction to encapsulate something like LRU, the cache has to record when an item is read from cache. Since `lookup` returns the actual value in cache and we can't perform side-effects, another operation is required to update the cache to track LRU.

I meant to add a comment in the code, but this interface is modeled directly after the [caching protocol in clojure.core.cache](http://clojure.github.io/core.cache/)